wasmd/services/scaffolder/scaffolder.go

```diff
@@ -0,0 +1,87 @@
+// Package scaffolder initializes Ignite CLI apps and modifies existing ones
+// to add more features in a later time.
+package scaffolder
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/ignite/cli/ignite/pkg/cmdrunner"
+	"github.com/ignite/cli/ignite/pkg/cmdrunner/step"
+	"github.com/ignite/cli/ignite/pkg/cosmosanalysis"
+	"github.com/ignite/cli/ignite/pkg/cosmosver"
+	"github.com/ignite/cli/ignite/pkg/gocmd"
+	"github.com/ignite/cli/ignite/pkg/gomodulepath"
+	"github.com/ignite/cli/ignite/version"
+)
+
+const (
+	wasmImport = "github.com/CosmWasm/wasmd"
+)
+
+// Scaffolder is Ignite CLI app scaffolder.
+type Scaffolder struct {
+	// Version of the chain
+	Version cosmosver.Version
+
+	// path of the app.
+	path string
+
+	// modpath represents the go module path of the app.
+	modpath gomodulepath.Path
+}
+
+// New creates a new scaffold app.
+func New(appPath string) (Scaffolder, error) {
+	path, err := filepath.Abs(appPath)
+	if err != nil {
+		return Scaffolder{}, err
+	}
+
+	modpath, path, err := gomodulepath.Find(path)
+	if err != nil {
+		return Scaffolder{}, err
+	}
+
+	ver, err := cosmosver.Detect(path)
+	if err != nil {
+		return Scaffolder{}, err
+	}
+
+	// Make sure that the app was scaffolded with a supported Cosmos SDK version
+	if err := version.AssertSupportedCosmosSDKVersion(ver); err != nil {
+		return Scaffolder{}, err
+	}
+
+	if err := cosmosanalysis.IsChainPath(path); err != nil {
+		return Scaffolder{}, err
+	}
+
+	s := Scaffolder{
+		Version: ver,
+		path:    path,
+		modpath: modpath,
+	}
+
+	return s, nil
```

Comment:
Could optionally also be written to be more compact as:

```suggestion
	return Scaffolder{
		Version: ver,
		path:    path,
		modpath: modpath,
	}, nil
```